### PR TITLE
Add coverage support

### PR DESCRIPTION
**Pull request by sylvestre**
The windows support for coverage fails with:
```
error[E0786]: found invalid metadata files for crate `vte_generate_state_changes`
Error:  --> C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\vte-0.10.1\src\table.rs:5:5
  |
5 | use vte_generate_state_changes::generate_state_changes;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: failed to decompress metadata: \\?\D:\a\sccache\sccache\target\debug\deps\vte_generate_state_changes-aafeba11e2d6a862.dll
```

### CODE REVIEW

**Comment by sylvestre** [comment]
Works on my fork - https://codecov.io/gh/sylvestre/sccache/tree/bb258858a9adf13847f6abe653498906d5058e9c - coverage of  56.80% 
but fails here because of:

`codecov/codecov-action@v1 is not allowed to be used in mozilla/sccache. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub or match the following: !/mozilla/**, !mozilla/**, ./**, 10up/wpcs-action@*, aws-actions/*, docker/*, pypa/gh-action-pypi-publish@v1.4.2, slackapi/slack-github-action@*.
`

**Comment by sylvestre** [comment]
Yeah, coverage works now!
For mozilla staff, see: https://bugzilla.mozilla.org/show_bug.cgi?id=1743848

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -155,6 +155,71 @@ jobs:
...
+    continue-on-error: ${{ matrix.allow_failure || false }}
+    timeout-minutes: 15
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-18.04
+            rustc: nightly
+            allow_failure: true
+            extra_args: --features=unstable
```
Comment (mitchhentges) [inline .github/workflows/ci.yml:170]: Why is `--features=unstable` needed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -155,6 +155,71 @@ jobs:
...
+          - os: macOS-10.15
+            rustc: nightly
+# Disable for now Windows as it fails with:
```
Comment (mitchhentges) [inline .github/workflows/ci.yml:173]: Mild English nit: "Disable on Windows for now as it fails with:"

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -155,6 +155,71 @@ jobs:
...
+# found invalid metadata files for crate `vte_generate_state_changes`
```
Comment (mitchhentges) [inline .github/workflows/ci.yml:174]: Can you refer (ideally here, or the commit message) to the upstream bug report that's blocking us from enabling this?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -155,6 +155,71 @@ jobs:
...
+        shell: bash
+        run: |
+          ## Generate coverage data
+          COVERAGE_REPORT_DIR="target/debug"
+          COVERAGE_REPORT_FILE="${COVERAGE_REPORT_DIR}/lcov.info"
+          # GRCOV_IGNORE_OPTION='--ignore build.rs --ignore "/*" --ignore "[a-zA-Z]:/*"' ## `grcov` ignores these params when passed as an environment variable (why?)
+          # GRCOV_EXCLUDE_OPTION='--excl-br-line "^\s*((debug_)?assert(_eq|_ne)?!|#\[derive\()"' ## `grcov` ignores these params when passed as an environment variable (why?)
+          mkdir -p "${COVERAGE_REPORT_DIR}"
+          # display coverage files
+          grcov . --output-type files --ignore build.rs --ignore "/*" --ignore "[a-zA-Z]:/*" --excl-br-line "^\s*((debug_)?assert(_eq|_ne)?!|#\[derive\()" | sort --unique
```
Comment (mitchhentges) [inline .github/workflows/ci.yml:209]: > `^\s*((debug_)?assert(_eq|_ne)?!|#\[derive\()"`

😳

**Review by mitchhentges — APPROVED** [review]
Looks good to me, though I'm trusting that you've tested this - I'd prefer to avoid doing a more methodical verification of your patch here.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (sylvestre) [reply .github/workflows/ci.yml:170]: to match the regular CI run

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sylvestre) [reply .github/workflows/ci.yml:173]: Merci!

File: .github/workflows/ci.yml (hunk 4 above)
Comment (sylvestre) [reply .github/workflows/ci.yml:209]: I took it from another project :p

**Pull request merged**